appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmMetadataCache.java

```diff
@@ -61,10 +60,8 @@ public String putMetadata(String path,String value) {
    * @return the attribute's string value or null if retrieval has failed.
    */
   public String getMetadata(String path) {
-    synchronized (cache) {
-      if (cache.containsKey(path)) {
-        return cache.get(path);
-      }
+    if (cache.containsKey(path)) {
```

Comment:
Oh, my fault, I missed this (2 operations that are synchronized together).
However, we can still do it by
(1) If `null` is not a possible value then just skip the `containsKey` and only use `get` and return
if not `null`.

(2) if `null` is possible use `getOrDefault` where default value is define in this class as `private static final Object NO_VALUE = new Object();` and if `value != NO_VALUE` then return it.